Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.4
+
+ENV GOPATH /go
+ENV PATH ${PATH}:/go/bin
+ENV TARGET /source
+ENV GOPATH_TARGET /go/src/github.com/docker/docker-network
+
+ENV SUBSYSTEMS namespace
+
+RUN go get -u github.com/kr/godep
+RUN mkdir ${TARGET}
+RUN mkdir -p $(dirname ${GOPATH_TARGET}) && \ 
```

Comment:
That's what the makefile is for; notice that it sets up a volume that
links the source in directly.

I've got a few projects using this strategy (without godep) and it works
very well.

On Mon, 2015-01-05 at 18:26 -0800, Alexander Morozov wrote:
> In Dockerfile:
> 
> > @@ -0,0 +1,17 @@
> > +FROM golang:1.4
> > +
> > +ENV GOPATH /go
> > +ENV PATH ${PATH}:/go/bin
> > +ENV TARGET /source
> > +ENV GOPATH_TARGET /go/src/github.com/docker/docker-network
> > +
> > +ENV SUBSYSTEMS namespace
> > +
> > +RUN go get -u github.com/kr/godep
> > +RUN mkdir ${TARGET}
> > +RUN mkdir -p $(dirname ${GOPATH_TARGET}) && \ 
> 
> I think it's probably not cool to assume that source mounted
> to /source. Then it won't be buildable by just docker build
> https://github.com/docker/docker-network. Ideas is to COPY source or
> just go get inside Dockerfile.
> 
> —
> Reply to this email directly or view it on GitHub.
> 
>